internal/gbimporter/gbimporter.go

```diff
@@ -81,7 +91,21 @@ func (i *importer) ImportFrom(path, srcDir string, mode types.ImportMode) (*type
 	def.SplitPathList = i.splitPathList
 	def.JoinPath = i.joinPath
 
-	return i.underlying.ImportFrom(path, srcDir, mode)
+	pkg, err := i.underlying.ImportFrom(path, srcDir, mode)
+	if err != nil {
+		return nil, err
+	}
+
+	// add the package to the cache
+	i.cache[path] = pkg
+	// delete the pakcage after ttl expires
+	timer := time.NewTimer(time.Duration(i.ttl) * time.Minute)
+	go func() {
```

Comment:
Does it really make sense to spawn a new goroutine for each and every imported package that lives for 60 minutes?

How about including the time the package has been added to the cache and check if it's still in the `ttl`

(sample code, probably not 100% correct)
```go
if c, ok := i.cache[path]; ok {
    if time.Since(c.Added) < time.Duration(i.ttl)*60 {
        delete(i.cache, path)
    } else {
        return c.Pkg, nil
    }
}

// .....

i.cache[path] = cachedPackage{
    Added: time.Now(),
    Pkg: pkg,
}
```

Reply:
Good idea removed the go routines and store the ttl as a unix time and compare it